learner_pathway_progress/utilities.py

```diff
@@ -0,0 +1,188 @@
+        log.exception(
+            f"{PATHWAY_LOGS_IDENTIFIER}: Encountered HTTPError {err} while getting snapshot of pathway: {pathway_uuid}"
+            f" from discovery."
+        )
+
+    return result
+
+
+def update_learner_pathway_progress(user_id, course_id):
+    """Update progress of all pathways linked with this course_id."""
+    user = User.objects.filter(id=user_id).first()
+    log.info(
+        f"{PATHWAY_LOGS_IDENTIFIER}: Update pathways associated with course: {course_id} for user {user_id}"
+    )
+    pathways_linked_with_course = learner_pathways_associated_with_course(user, course_id)
+    if pathways_linked_with_course:
+        for learner_pathway_uuid in pathways_linked_with_course:
+            pathway_snapshot = get_pathway_snapshot(user, learner_pathway_uuid)
+            if pathway_snapshot:
+                LearnerPathwayMembership.objects.get_or_create(
+                    user=user,
+                    learner_pathway_uuid=learner_pathway_uuid
+                )
+                pathway_progress, _ = LearnerPathwayProgress.objects.get_or_create(
+                    user=user,
+                    learner_pathway_uuid=learner_pathway_uuid,
+                )
+                pathway_progress.learner_pathway_progress = json.dumps(pathway_snapshot)
+                pathway_progress.save()
+                pathway_progress.update_pathway_progress()
+                log.info(
+                    f"{PATHWAY_LOGS_IDENTIFIER}: Pathway:{learner_pathway_uuid} updated associated "
+                    f"with course: {course_id}"
+                )
+
+
+def get_all_learner_pathways(user_id):
+    """Return the all pathways."""
+    user = User.objects.filter(id=user_id).first()
+    client = create_catalog_api_client(user)  # pylint: disable=assignment-from-none
+    learner_pathways_url = urljoin(f"{get_catalog_api_base_url()}/", "learner-pathway/")
+    learner_pathways = None
+    try:
+        response = client.get(learner_pathways_url)
+        response.raise_for_status()
+        learner_pathways = response.json()
+        learner_pathways_count = len(learner_pathways['results'])
+        log.info(
+            f"{PATHWAY_LOGS_IDENTIFIER}: Total Learner Pathways are : {learner_pathways_count}"
+        )
+    except HTTPError as err:
+        log.exception(
+            f"{PATHWAY_LOGS_IDENTIFIER}: Encountered HTTPError {err} while getting all pathways from discovery. "
+        )
+    return learner_pathways
+
+
+def back_fill_learner_pathway_membership_and_progress(pathways):
+    """Update the pathway progress and created membership for user if not already exist."""
+    enrollments_count = EnterpriseCourseEnrollment.objects.all().count()
+    log.info(
+        f"{PATHWAY_LOGS_IDENTIFIER}: Total Enterprise Course Enrollment Count: {enrollments_count}"
+    )
+    for offset in range(0, enrollments_count, CHUNK_SIZE):
+        ent_course_enrollments = EnterpriseCourseEnrollment.objects.all()[
```

Comment:
You could use the queryset iterator with chunking or the Paginator https://nextlinklabs.com/insights/django-big-data-iteration

Reply:
Thank you. I've changed it to paginator.